db/skipped_sequence.go

```diff
@@ -181,34 +279,106 @@ func (s *SkippedSequenceSlice) removeSeq(x uint64) error {
 		// if not a range, we can just remove the single entry from the slice
 		s.list = slices.Delete(s.list, index, index+1)
 		return nil
+	} else {
+		// range entry handling
+		rangeElem := s.list[index]
+		numSequences := rangeElem.getNumSequencesInEntry()
+		if numSequences == 1 {
+			// simple delete case
+			s.list = slices.Delete(s.list, index, index+1)
+			return nil
+		}
+
+		// if x == startSeq set startSeq+1, if x == lastSeq set lastSeq-1
+		if rangeElem.getStartSeq() == x {
+			rangeElem.setStartSeq(x + 1)
+			return nil
+		}
+		if rangeElem.getLastSeq() == x {
+			rangeElem.setLastSeq(x - 1)
+			return nil
+		}
```

Comment:
So the original approach was if the above code lines makes a range of 1, we were going to delete this and insert a new single entry in its place. Adam suggested we could avoid extra allocations by just altering the range to be one sequence. It seems to work well! 
With regards to it running through `isRange()`, it would still return true for being a range entry but I think this has no functional impact. Where `isRange()` is used in binary search function, it will still find the sequence if the range is length 1. Also for `removeSeq` we have a check for 1 sequence (above this code) that will just delete the entry.  The only other way this is used is for adding new entries in `PushSkippedSequenceEntry` which I feel doesn't have functional impact.